src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -585,6 +505,93 @@ def set_up_update_containerapp_yaml(self, name, file_name):
+                            c["resources"]["cpu"] = self.get_argument_cpu()
+                        if self.get_argument_memory() is not None:
+                            c["resources"]["memory"] = self.get_argument_memory()
+                    else:
+                        c["resources"] = {
+                            "cpu": self.get_argument_cpu(),
+                            "memory": self.get_argument_memory()
+                        }
+                if self.get_argument_secret_volume_mount() is not None:
+                    self.new_containerapp["properties"]["template"]["volumes"] = self.containerapp_def["properties"]["template"]["volumes"]
+                    if "volumeMounts" not in c or not c["volumeMounts"]:
+                        # if no volume mount exists, create a new volume and then mount
+                        volume_def = VolumeModel
+                        volume_mount_def = VolumeMountModel
+                        volume_def["name"] = _generate_secret_volume_name()
+                        volume_def["storageType"] = "Secret"
+
+                        volume_mount_def["volumeName"] = volume_def["name"]
+                        volume_mount_def["mountPath"] = self.get_argument_secret_volume_mount()
+
+                        if "volumes" not in self.new_containerapp["properties"]["template"] or self.new_containerapp["properties"]["template"]["volumes"] is None:
+                            self.new_containerapp["properties"]["template"]["volumes"] = [volume_def]
+                        else:
+                            self.new_containerapp["properties"]["template"]["volumes"].append(volume_def)
+                        c["volumeMounts"] = [volume_mount_def]
+                    else:
+                        if len(c["volumeMounts"]) > 1:
+                            raise ValidationError(
+                                "Usage error: --secret-volume-mount can only be used with a container that has a single volume mount, to define multiple volumes and mounts please use --yaml")
+                        else:
+                            # check that the only volume is of type secret
+                            volume_name = c["volumeMounts"][0]["volumeName"]
+                            for v in self.new_containerapp["properties"]["template"]["volumes"]:
+                                if v["name"].lower() == volume_name.lower():
+                                    if v["storageType"] != "Secret":
+                                        raise ValidationError(
+                                            "Usage error: --secret-volume-mount can only be used to update volume mounts with volumes of type secret. To update other types of volumes please use --yaml")
+                                    break
+                            c["volumeMounts"][0]["mountPath"] = self.get_argument_secret_volume_mount()
+            return updating_existing_container
```

Comment:
should always `return updating_existing_container`, currently you only return updating_existing_container in the `For`